splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/GroupByCorreltionIT.java

```diff
@@ -0,0 +1,147 @@
+            Assert.assertEquals("E1", resultSet.getString(1));Assert.assertEquals("P4", resultSet.getString(2));Assert.assertEquals(2, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E1", resultSet.getString(1));Assert.assertEquals("P5", resultSet.getString(2));Assert.assertEquals(2, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E1", resultSet.getString(1));Assert.assertEquals("P6", resultSet.getString(2));Assert.assertEquals(2, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E2", resultSet.getString(1));Assert.assertEquals("P1", resultSet.getString(2));Assert.assertEquals(4, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E2", resultSet.getString(1));Assert.assertEquals("P2", resultSet.getString(2));Assert.assertEquals(4, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E3", resultSet.getString(1));Assert.assertEquals("P2", resultSet.getString(2));Assert.assertEquals(5, resultSet.getInt(3));
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E4", resultSet.getString(1));Assert.assertEquals("P2", resultSet.getString(2));resultSet.getInt(3); Assert.assertTrue(resultSet.wasNull());
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E4", resultSet.getString(1));Assert.assertEquals("P4", resultSet.getString(2));resultSet.getInt(3); Assert.assertTrue(resultSet.wasNull());
+            Assert.assertTrue(resultSet.next());
+            Assert.assertEquals("E4", resultSet.getString(1));Assert.assertEquals("P5", resultSet.getString(2));resultSet.getInt(3); Assert.assertTrue(resultSet.wasNull());
+            Assert.assertFalse(resultSet.next());
+        }
+    }
+
+    @Test
+    public void ssqWithoutCorrelationOnGroupByColumnThrows() throws Exception {
+        try(ResultSet resultSet = methodWatcher.executeQuery("SELECT oc1, (SELECT MAX(ic1) FROM INNER_TABLE WHERE ic2 = outer_table.oc2) FROM OUTER_TABLE GROUP BY oc1 order by oc1 asc")) {
+            Assert.fail("expected exception containing message: The SELECT list of a non-grouped query contains at least one invalid expression. When the SELECT list contains at least one " +
+                    "aggregate then all entries must be valid aggregate expressions");
+        } catch(Exception se) {
+            Assert.assertTrue(se instanceof SQLException);
+            Assert.assertEquals("42Y29", ((SQLException)se).getSQLState());
+            Assert.assertTrue(se.getMessage().contains("The SELECT list of a non-grouped query contains at least one invalid expression. When the SELECT list contains at least one aggregate " +
+                    "then all entries must be valid aggregate expressions"));
+        }
+    }
+}
```

Comment:
In addition, could you test a non-correlated Scalar subquery case? Also a case where the Scalar subquery does not contain aggregation?

Reply:
The test for a non-correlated SSQ case is added to `GroupByOrderByIT` test suite, I also added here a negative test for a scenario where subquery does not contain aggregation as suggested.